### ISSUE TITLE
The short switch '-p' is not defined in the switch mappings

### ISSUE THREAD

**Issue by zhshqzyc** [aspnet/aspnet-docker#139]
My Dockerfile in an asp.net core 1.0 mvc project is 

```
FROM microsoft/aspnet:1.0.0-rc1-update1
ADD . /app
WORKDIR /app/approot
EXPOSE 5004
ENTRYPOINT ["dnx", "-p", "project.json", "web"]
```
Then in docker quick terminal.  Build the image was successfully. The command is
`docker build -t hellodocker:0.1.0  .`

![1](https://cloud.githubusercontent.com/assets/7240266/14502769/faeccef4-017a-11e6-8698-13302cd74e02.PNG)

However when I run the command
`docker run hellodocker:0.1.0 -p 5004`
I got the exception, 

![exception](https://cloud.githubusercontent.com/assets/7240266/14502790/17126e7c-017b-11e6-9627-43ed898010e1.PNG)

So what is wrong?

**Comment by friism** [comment]
@zhshqzyc I don't think you have to pass `-p` to the docker invocation

**Comment by zhshqzyc** [comment]
I removed the -p from the run command. The error is gone and it shows 
Now listening on `localhost:5000`.

I typed the url in the browser.
`Firefox can't establish a connection to the server at localhost:5000.`

**Comment by friism** [comment]
@zhshqzyc 

```
C:\code\aspnet-home\samples\1.0.0-rc1-update1\HelloMvc [dev ≡]> docker build -t aspnet-mvc .
...
docker run -p 5000:5000 aspnet-mvc --server.urls http://0.0.0.0:5000
...
```

The app will then be listening on port 5000.

**Comment by zhshqzyc** [comment]
@friism 

See the error by the code.

![5000](https://cloud.githubusercontent.com/assets/7240266/14504403/b19537de-0182-11e6-8d36-354697d26cf1.PNG)

**Comment by friism** [comment]
@zhshqzyc I don't know what's the layout of your project, so I can't help.

The example I posted shows running this: https://github.com/aspnet/Home/tree/dev/samples/1.0.0-rc1-update1/HelloMvc

**Comment by zhshqzyc** [comment]
@friism .

Okay, I downloaded the project from [ https://github.com/aspnet/Home/tree/dev/samples/1.0.0-rc1-update1/HelloMvc](url)

And I tried to follow your command. The only difference is the original project uses the port 5004 instead of 5000.
The Dockerfile is
```
FROM microsoft/aspnet:1.0.0-rc1-update1
COPY . /app/
WORKDIR /app
RUN ["dnu", "restore"]
EXPOSE 5004
ENTRYPOINT ["dnx", "-p", "project.json", "web"]
```
The partial project.json file is
```
 "commands": {
    "web": "Microsoft.AspNet.Server.Kestrel --server.urls http://*:5004"
  },
```
So the build command is `docker build -t aspnet-mvc .`
What is the run command?
I use `docker run -p 5004 aspnet-mvc --server.urls http://0.0.0.0:5004`
Am I right? But I open the `http://0.0.0.0.0:5004`, 

> This site can’t be reached
